analysis/generate_report.py

```diff
@@ -143,6 +144,40 @@ def get_experiment_data(experiment_names, main_experiment_name,
     return experiment_df, description
 
 
+def filter_experiments(all_experiment_names):
+    """Helper function that reads data from disk or from the database. Returns
+    a list of old experiments of the same type as the main experiment."""
+    main_experiment_name, prev_experiment_names = all_experiment_names[
```

Comment:
Set these variables on different lines.
Also set them closer to where they are used.